src/Filesystem/Feature/TransformImageUrl/GlideTransformImageUrl.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace Zenstruck\Filesystem\Feature\TransformImageUrl;
+
+use League\Glide\Urls\UrlBuilder;
+use Zenstruck\Filesystem\Feature\TransformImageUrl;
+use Zenstruck\Filesystem\Node\File\Image;
+use Zenstruck\Uri;
+
+/**
+ * @author Jakub Caban <kuba.iluvatar@gmail.com>
+ */
+class GlideTransformImageUrl implements TransformImageUrl
+{
+    public function __construct(private UrlBuilder $urlBuilder)
+    {
+    }
+
+    public function transformUrlFor(Image $image, mixed $options = []): Uri
+    {
+        return Uri::new($this->urlBuilder->getUrl($image->path(), $options));
```

Comment:
What do you think about this:

```suggestion
        $options = match(true) {
            \is_string($options) => ['p' => $options], // is glide "preset"
            \is_array($options) && !is_array_list($options) => $options, // is standard glide parameters
            \is_array($options) => ['p' => \implode(',', $options)], // is array of "presets"
            default => throw new \LogicException('invalid options... todo'),
        }

        return Uri::new($this->urlBuilder->getUrl($image->path(), $options));
```

This gives the ability to use [Glide presets](https://glide.thephpleague.com/1.0/config/defaults-and-presets/) easily.

Both Imagine and Intervention have the concept of _filters_ which you can use to name and wrap _multiple_ manipulations. The reason I chose `mixed $options` was to make it easier to pass a single string (the filter) to `Image::transformUrl()`: `$image->transformUrl('square_200')`.